### PR TITLE
GPT- 48, 65, 102 frontend components and display messages for authentication

### PR DESCRIPTION
**Pull request by neselus**
Updated the LoginForm and RegistrationForm for authentication. Designed the front-end components, implemented necessary display messages, and implemented error message for password requirements. 

It is currently in two big files, after the review breaking the code into smaller components might be needed.

### CODE REVIEW

File: crushit/.env.defaults
Comment (tanner666) [inline crushit/.env.defaults:1]: move the firebase api key stuff into the .env file. The .env.defaults is the "public" version of .env file, which is used for non-confidential stuff like setting up local databse. The .env file is for private/confidential stuff, like API keys and external db credentials

**Review by tanner666 — CHANGES_REQUESTED** [review]
move the api keys to the private .env file

File: crushit/.env.defaults
Comment (neselus) [reply crushit/.env.defaults:1]: Deleted the firebase api key from the .env.default file

**Review by tanner666 — APPROVED** [review]
looks good, just make sure to add unit tests

**Pull request merged**